### ISSUE TITLE
L010 L014 Non-alphabetic characters refute lowercase policy

### ISSUE THREAD

**Issue by julianazhu** [sqlfluff/sqlfluff#3206]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

We have the following rule set in our `.sqlfluff` and are using dbt v0.21
```
# non-keyword capitalization
[sqlfluff:rules:L014]
capitalisation_policy = lower
```
When compiling an 'ephemeral' dbt model, the model is created with the identifier `__dbt__cte__model_name_here`. The underscore at the beginning of the identifier refutes the lowercase policy and causes all subsequent identifiers to violate rule L014. 

### Expected Behaviour

Underscores and other non-alphabetic characters will not trigger `L014 | Unquoted identifiers must be consistently capitalised.` when applying the lowercase policy when all other identifiers follow the lowercase policy.

### Observed Behaviour

`L014 | Unquoted identifiers must be consistently capitalised.` was triggered on every line with an identifier and this was not resolved by lowercasing every word in the entire model. 

Terminal output:
```sh
$ sqlfluff lint models/test.sql
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models/test.sql] FAIL
L:   3 | P:   9 | L014 | Unquoted identifiers must be consistently capitalised.
L:   3 | P:  13 | L014 | Unquoted identifiers must be consistently capitalised.
L:   4 | P:  11 | L014 | Unquoted identifiers must be consistently capitalised.
L:   4 | P:  15 | L014 | Unquoted identifiers must be consistently capitalised.
L:   5 | P:  45 | L014 | Unquoted identifiers must be consistently capitalised.
L:   5 | P:  50 | L014 | Unquoted identifiers must be consistently capitalised.
L:   5 | P:  53 | L014 | Unquoted identifiers must be consistently capitalised.
```

### How to reproduce

1. Set the ` capitalisation_policy = lower ` in `.sqlfluff`

2. Create a dbt model such as (this one uses Snowflake syntax):

```sql
with __test as (
    select
        foo.a
        , foo.b
    from (values (1, 'one'), (2, 'two')) as foo (a, b)
)

select * from __test
```

3. Run `sqlfluff lint` on the model

### Dialect

Snowflake

### Version

Python 3.8.12
sqlfluff, version 0.6.9

### Configuration

```
[sqlfluff]
verbose = 0
nocolor = False
dialect = snowflake
templater = dbt
# The rules reference https://docs.sqlfluff.com/en/stable/rules.html
rules = L001,L004,L005,L008,L009,L010,L011,L013,L014,L015,L016,L017,L018,L019,L020,L021,L023,L024,L025,L026,L027,L028,L030,L036,L038,L039,L040,L041,L045,L046,L048
exclude_rules = L002,L003,L006, L007,L012, L022,L029
output_line_length = 90

[sqlfluff:indentation]
indented_joins = False

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 120
indent_unit = space

# keyword capitalization
[sqlfluff:rules:L010]
capitalisation_policy = lower

# all columns must have explicit AS alias even a single scalar column
[sqlfluff:rules:L013]
allow_scalar = False

# non-keyword capitalization
[sqlfluff:rules:L014]
capitalisation_policy = lower

# line length
[sqlfluff:rules:L016]
max_line_length = 120
tab_space_size = 4
indent_unit = space

[sqlfluff:rules:L019]
comma_style = leading

[sqlfluff:rules:L028]
single_table_references = consistent

[sqlfluff:rules:L029]
only_aliases = true

# function capitalization
[sqlfluff:rules:L030]
capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
